src/libmongoc/CMakeLists.txt

```diff
@@ -708,6 +707,15 @@ set (STATIC_LIBRARIES
    ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
 )
 
+# utf8proc configuration
+find_package(Utf8Proc REQUIRED)
+if ("${USE_BUNDLED_UTF8PROC}" STREQUAL "TRUE")
+   list(APPEND SOURCES $<TARGET_OBJECTS:utf8proc_obj>)
```

Comment:
Propagating object files into the mongoc static libraries may wreak havoc if users wish to use a different utf8proc in their application. This could be mitigated with linker tricks, but I think a cleaner CMake-based approach may be possible and simpler to maintain. I would not consider this a blocker for now, and can be postponed.